### ISSUE TITLE
Astro components don't work if passed as children to hydrated svelte components

### ISSUE THREAD

**Issue by JulioGrajales** [withastro/astro#11612]
### Astro Info

```block
Astro                    v4.13.1
Node                     v22.4.1
System                   Windows (x64)
Package Manager          pnpm
Output                   static
Adapter                  none
Integrations             @astrojs/tailwind
                         @astrojs/svelte
                         @astrojs/mdx
                         @astrojs/sitemap
```

### Describe the Bug

If you pass an astro component to a svelte component slot and if you hydrate that svelte component the astro component stops working after 2 clicks or may not work at all.

https://github.com/user-attachments/assets/0bc1af80-bd86-4123-9084-88d52b6e3bfb

### What's the expected result?

The astro component should work when passed as a slot to a svelte component

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-iqe45d?file=src%2Fpages%2Findex.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
I'm not sure if this is an Astro issue. The problem is that the script is hoisted as a `<script type="module">` and will run once. At the same time, Astro will also try to hydrate the Svelte island, which requires a fetch and hydrate that by the time it's done, the button script has already run.

However, the button script will capture the non-hydrated button first, and then Svelte hydrates and replaces the button instance, causing the reference to be lose.

In this case, I think you need to anticipate that buttons could be loaded later by modifying your script, using `MutationObserver` perhaps or custom elements. You may also have the button loaded later if the Svelte component can toggle its display, so it's not only related to hydration.

**Comment by bluwy** [comment]
Yeah I think we can clarify in our docs about the limitations of the examples. They currently don't work well when the HTML elements they're querying appears later.
